src/main/java/fr/esgi/al5_2/Tayarim/controllers/AuthController.java

```diff
@@ -15,88 +16,172 @@
-            String errorMessage = error.getDefaultMessage();
-            errors.add(errorMessage);
-        });
-
-        errorMapping.put("errors", errors);
-
-        return errorMapping;
-    }
-
-    @ResponseStatus(HttpStatus.NOT_FOUND)
-    @ExceptionHandler({ProprietaireNotFoundException.class})
-    public Map<String, List<String>> proprietaireNotFoundException(ProprietaireNotFoundException ex){
-        return mapException(ex);
-    }
-
-    @ResponseStatus(HttpStatus.UNAUTHORIZED)
-    @ExceptionHandler({TokenExpireOrInvalidException.class})
-    public Map<String, List<String>> tokenExpireOrInvalidException(TokenExpireOrInvalidException ex){
-        return mapException(ex);
-    }
-
-    private Map<String, List<String>> mapException(RuntimeException exception){
-        ArrayList<String> errors = new ArrayList<>();
-        Map<String, List<String>> errorMapping = new HashMap<>();
-        errors.add(exception.getMessage());
-
-        errorMapping.put("errors", errors);
-
-        return errorMapping;
-    }
+    return authHeader.substring(7);
+  }
+
+  /**
+   * Gère les exceptions de validation des arguments de méthode.
+   *
+   * @param ex L'exception capturée.
+   * @return Une carte des erreurs.
+   */
+  @ResponseStatus(HttpStatus.BAD_REQUEST)
+  @ExceptionHandler(MethodArgumentNotValidException.class)
+  public Map<String, List<String>> handleValidationExceptions(MethodArgumentNotValidException ex) {
+    ArrayList<String> errors = new ArrayList<>();
+    Map<String, List<String>> errorMapping = new HashMap<>();
+    ex.getBindingResult().getAllErrors().forEach((error) -> {
+      String errorMessage = error.getDefaultMessage();
+      errors.add(errorMessage);
+    });
+
+    errorMapping.put("errors", errors);
+
+    return errorMapping;
+  }
+
+  /**
+   * Gère les exceptions lorsque l'utilisateur n'est pas trouvé.
+   *
+   * @param ex L'exception capturée.
+   * @return Une carte des erreurs.
+   */
+  @ResponseStatus(HttpStatus.NOT_FOUND)
+  @ExceptionHandler({UtilisateurNotFoundException.class})
+  public Map<String, List<String>> utilisateurNotFoundException(UtilisateurNotFoundException ex) {
+    return mapException(ex);
+  }
+
+  /**
+   * Gère les exceptions lorsque le proprietaire n'est pas trouvé.
+   *
+   * @param ex L'exception capturée.
+   * @return Une carte des erreurs.
+   */
+  @ResponseStatus(HttpStatus.NOT_FOUND)
+  @ExceptionHandler({ProprietaireNotFoundException.class})
+  public Map<String, List<String>> proprietaireNotFoundException(ProprietaireNotFoundException ex) {
+    return mapException(ex);
+  }
+
+  /**
+   * Gère les exceptions lorsque l'administrateur n'est pas trouvé.
+   *
+   * @param ex L'exception capturée.
+   * @return Une carte des erreurs.
+   */
+  @ResponseStatus(HttpStatus.NOT_FOUND)
+  @ExceptionHandler({AdministrateurNotFoundException.class})
+  public Map<String, List<String>> administrateurNotFoundException(AdministrateurNotFoundException ex) {
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 104).